statsmodels/regression/mixed_linear_model.py

```diff
@@ -2382,6 +2382,20 @@ def summary(self, yname=None, xname_fe=None, xname_re=None,
         info["Model:"] = "MixedLM"
         if yname is None:
             yname = self.model.endog_names
+        param_names = self.model.data.param_names[:]
+        n_fe_params = len(self.fe_params)
+        n_re_params = len(param_names) - len(self.fe_params)
+        if xname_fe is None:
+            xname_fe = param_names[:n_fe_params]
+        if len(xname_fe) != n_fe_params:
+            raise ValueError("xname_fe should be a list of length %d" % n_fe_params)
+        if xname_re is None:
+            xname_re = param_names[n_fe_params:]
+        if len(xname_re) != n_re_params:
+            raise ValueError("xname_re should be a list of length %d" % n_re_params)
+        param_names[:n_fe_params] = xname_fe
+        param_names[n_fe_params:] = xname_re
```

Comment:
i think it would be better to change the logic here to only assign if xname_fe (by user) is not None, and don't change param_names if it is not needed
e.g
```
if xname_fe is not None:
  if len(xname_fe) != n_fe_params:
        raise ValueError("xname_fe should be a list of length %d" % n_fe_params)
  param_names[:n_fe_params] = xname_fe

if xname_re is not None:
...
```

Reply:
All good points, and dully noted for the naming convention.